src/pages/admin/Event/EventStats/MassUpdateUserStatus.js

```diff
@@ -0,0 +1,158 @@
+import React, {
+  useState
+} from "react";
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  TextField,
+  Select,
+  MenuItem,
+  DialogActions,
+  Button
+} from "@material-ui/core";
+import {
+  APPLICATION_STATUS
+} from "constants/_constants/eventStatsStatusFields";
+import {
+  fetchBackend
+} from "utils";
+
+const MassUpdateModal = ({
+  open,
+  onClose,
+  eventID,
+  eventYear,
+  refreshTable
+}) => {
+  const [emailList, setEmailList] = useState("");
+  const [selectedStatus, setSelectedStatus] = useState("");
+
+  const fetchFnameByEmail = async (email) => {
+    try {
+      console.log("calling fname");
+      const response = await fetchBackend(`/members/${email}/fname`, "GET");
+      console.log("call fname complete");
+      if (response.firstName) {
+        return response.firstName;
+      } else {
+        throw new Error(`First name not found for email: ${email}`);
+      }
+    } catch (error) {
+      console.error("Error fetching fname for email:", email, error);
+      throw new Error(`Failed to fetch fname for email: ${email}`);
+    }
+  };
+
+  const handleSubmit = async () => {
+    const emails = emailList.split(",").map((email) => email.trim());
+
+    try {
+      const updates = await Promise.all(emails.map(async (email) => {
+        const fname = await fetchFnameByEmail(email);
```

Comment:
This probably works, but for future reference, having this inner async call is prob pretty slow because you're doing this one function for every email. You could pass that responsibility to the backend so that the backend is responsible for finding the proper fname and so the FE just needs to send the list of emails.

Reply:
Ohhh okok thanks. I will investigate that

Reply:
no longer using inner async